arbitrator/prover/src/merkle.rs

```diff
@@ -393,20 +371,39 @@ impl Merkle {
             );
         }
         let mut layers = self.layers.lock();
+        let start = layers.data[0].len();
         let mut layer_size = new_len;
         for (layer_i, layer) in layers.data.iter_mut().enumerate() {
             layer.resize(layer_size, *empty_hash_at(self.ty, layer_i));
             layer_size = max(layer_size >> 1, 1);
         }
-        let start = layers.data[0].len();
         for i in start..new_len {
```

Comment:
nit: I think it'd be a bit more efficient and possibly a bit clearer to do something like this:

```rs
// I think this will always set either one value or no values at all, so you could make this into an if statement that calls set, either way
layers.dirty_leaf_parents[(start >> 1)..].fill(true);
// then once we've set the existing value if necessary, we can fill in the new values with resize, which will more efficiently set them and do a single reallocation if necessary
layers.dirty_leaf_parents.resize(new_len >> 1, true);
```

Reply:
Done.